### PR TITLE
Update podSecurityContext to support random users IDs and fsGroup

### PR DESCRIPTION
**Pull request by afalhambra-hivemq**
https://hivemq.kanbanize.com/ctrl_board/22/cards/19506/details/

### CODE REVIEW

File: charts/hivemq-platform-operator/crds/hivemq-platforms.hivemq.com-v1.yml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# Generated by Fabric8 CRDGenerator, manual edits might get overwritten!
```
Comment (afalhambra-hivemq) [inline charts/hivemq-platform-operator/crds/hivemq-platforms.hivemq.com-v1.yml:1]: if we don't update master branch of helm-chart with the CRD changes, the `hivemq-k8s-testsuite` repo might fail in the IT when comparing the CRDs, since at the moment this CRD will only be updated on the epic branch, and pipelines in the jenkins for other repos will use the master branch of the `helm-charts`, `hivemq-platform-operator`, and `hivemq-k8s-testsuite`.

File: charts/hivemq-platform-operator/templates/deployment.yml [hunk 2]
```diff
@@ -24,10 +24,12 @@ spec:
...
       securityContext:
-        runAsUser: {{ .Values.podSecurityContext.runAsUser }}
-        runAsGroup: {{ .Values.podSecurityContext.runAsUser }}
-        fsGroup: {{ .Values.podSecurityContext.runAsUser }}
         runAsNonRoot: {{ .Values.podSecurityContext.runAsNonRoot }}
+        runAsUser: {{ .Values.podSecurityContext.runAsUser | default 185 }}
+        runAsGroup: {{ .Values.podSecurityContext.runAsGroup | default 0 }}
+        {{- if .Values.podSecurityContext.fsGroup }}
+        fsGroup: {{ .Values.podSecurityContext.fsGroup }}
+        {{- end }}
```
Comment (afalhambra-hivemq) [inline charts/hivemq-platform-operator/templates/deployment.yml:32]: we cannot set this `fsGroup` to be the same as the `runAsGroup`. On Openshift this value will be a range of GIDs, different than the `runAsGroup`.

**Review by Donnerbart — APPROVED** [review]
Nice work, I hope we catched all edge cases :)

Could you please update the manifest files as well?

File: tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/single/HelmNonRootUserInstallPlatformIT.java [hunk 3]
```diff
@@ -1,25 +1,93 @@
...
+    void installPlatform_withDefaultNonRootUsers_hivemqRunning() throws Exception {
+        HELM_CHART_CONTAINER.installOperatorChart(OPERATOR_RELEASE_NAME,
+                "-f",
+                "/files/operator-non-root-user-values.yaml");
+        HELM_CHART_CONTAINER.installPlatformChart(PLATFORM_RELEASE_NAME,
+                "-f",
+                "/files/platform-non-root-user-values.yaml",
+                "--namespace",
+                PLATFORM_NAMESPACE);
+        K8sUtil.waitForHiveMQPlatformStateRunning(client, PLATFORM_NAMESPACE, PLATFORM_RELEASE_NAME);
```
Comment (mhofsche) [inline tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/single/HelmNonRootUserInstallPlatformIT.java:73]: Does it make sense to check the uid and gid here as well? 
Along the lines of `client.pods().inNamespace(namespace).withName(...).exec("stat -c \"%u %g\" /proc/1")` --> 185 0

Comment (afalhambra-hivemq) [reply tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/single/HelmNonRootUserInstallPlatformIT.java:85]: Good suggestion!
Added now for both tests.

**Review by mhofsche — APPROVED** [review]
